src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Lengths must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(`This is not a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.color = color;
+    this.radius = radius;
 
-export class Rectangle implements Figure {}
+    if (radius <= 0) {
+      throw new Error('Radius should be greater than 0');
+    }
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    this.color = color;
+    this.width = width;
+    this.height = height;
```

Comment:
The assignments `this.color = color;`, `this.width = width;`, and `this.height = height;` are redundant because you are already using `public` in the constructor parameters, which automatically assigns these values to the instance properties. You can safely remove these assignments.